### PR TITLE
chore: Bump Keycloak dependency, update realm configurations

### LINKED ISSUE #15379: Upgrade Keycload to 14.0.x
##### **Overview of the feature request**

See https://www.keycloak.org/2021/06/keycloak-1400-released.html

##### **Related issues or PR**

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by vishal423**
Closes #15379 
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/src/main/docker/config/realm-config/jhipster-realm.json.ejs [hunk 1]
```diff
@@ -69,6 +72,27 @@
         "containerId": "jhipster",
         "attributes": {}
       },
+      {
+        "id": "d2b73e7b-a2d7-40e9-8ebc-2af00454e8aa",
+        "name": "default-roles-jhipster",
```
Comment (Tcharl) [inline generators/server/templates/src/main/docker/config/realm-config/jhipster-realm.json.ejs:77]: Maybe a more descriptive role?
'service_account_role'?
Also, I don't see why uma is activated.

Comment (vishal423) [reply generators/server/templates/src/main/docker/config/realm-config/jhipster-realm.json.ejs:77]: `default-roles-jhipster` is an internal name (auto-generated) assigned to default roles in the `jhipster` realm. 
I haven't specifically enabled `uma_authorization` and it seems another default.

Unless there is a strong reason to change, I will prefer to not update default settings.

**Comment by vishal423** [comment]
@pascalgrimaud , bounty claimed https://opencollective.com/generator-jhipster/expenses/44316

**Comment by pascalgrimaud** [comment]
@vishal423 : approved

**Pull request merged**